src/pal/src/include/pal/context.h

```diff
@@ -277,6 +277,57 @@ inline void *FPREG_Xstate_Ymmh(const ucontext_t *uc)
 #define MCREG_Sp(mc)      ((mc).sp)
 #define MCREG_Pc(mc)      ((mc).pc)
 #define MCREG_Cpsr(mc)    ((mc).pstate)
+
+
+#define _ALIGN_UP_16(s) ((s + 15) & ~0xf)
+inline
+fpsimd_context* GetNativeSigSimdContext(native_context_t *mc, bool createIfNotPresent)
+{
+    fpsimd_context* fp = nullptr;
+
+    size_t size = 0;
+    _aarch64_ctx* record = reinterpret_cast<_aarch64_ctx *>(&mc->uc_mcontext.__reserved[size]);
+
+    while (record->size !=0)
```

Comment:
I seems we don't ever have to create the fpsimd_context, we can assume it is always present. Looking at the kernel source, it always puts it as the first extra context into the reserved area:
https://elixir.bootlin.com/linux/v5.1.15/source/arch/arm64/kernel/signal.c#L642

So I'd assert if it was not there.